cpp/FishHash.cpp

```diff
@@ -177,18 +177,18 @@ namespace FishHash {
 	}
 
 	void hash(uint8_t * output, const fishhash_context * ctx, const uint8_t * header, uint64_t header_size) noexcept {
-		hash512 seed; 
+		hash1024 seed; 
 	   
 		blake3_hasher hasher;
 		blake3_hasher_init(&hasher);
 		blake3_hasher_update(&hasher, header, header_size);
-		blake3_hasher_finalize(&hasher, seed.bytes, 64);
+		blake3_hasher_finalize(&hasher, seed.bytes, 128);
 				
 		const hash256 mix_hash = fishhash_kernel(*ctx, seed);
 	    
-		uint8_t final_data[sizeof(seed) + sizeof(mix_hash)];
-		std::memcpy(&final_data[0], seed.bytes, sizeof(seed));
-		std::memcpy(&final_data[sizeof(seed)], mix_hash.bytes, sizeof(mix_hash));
+		uint8_t final_data[sizeof(seed) / 2 + sizeof(mix_hash)];
+		std::memcpy(&final_data[0], seed.bytes, sizeof(seed) / 2);
```

Comment:
Not sure the reasoning for the first half of the final output being the seed. Now that the seed is longer, modifying this to only take the first half of the seed